pkg/collector/client.go

```diff
@@ -32,46 +60,104 @@ func (c *clientCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	ch <- prometheus.MustNewConstMetric(c.omadaClientConnectedTotal, prometheus.GaugeValue, float64(len(clients)),
-		site, client.SiteId)
+	totals := map[string]int{}
 
 	for _, item := range clients {
 		vlanId := fmt.Sprintf("%.0f", item.VlanId)
 		port := fmt.Sprintf("%.0f", item.Port)
+		connectionModeFmt := ""
 		if item.Wireless {
 			wifiMode := fmt.Sprintf("%.0f", item.WifiMode)
+			connectionModeFmt = FormatWifiMode(int(item.WifiMode))
+
+			CollectWirelessMetrics := func(desc *prometheus.Desc, valueType prometheus.ValueType, value float64) {
+				ch <- prometheus.MustNewConstMetric(desc, valueType, value,
+					item.HostName, item.Vendor, item.Ip, item.Mac, item.ApName, site, client.SiteId, "wireless", item.Ssid, wifiMode, connectionModeFmt)	
+			}
+			CollectWirelessMetrics(c.omadaClientSignalDbm, prometheus.GaugeValue, -item.SignalLevel)
+			CollectWirelessMetrics(c.omadaClientRssiDbm, prometheus.GaugeValue, item.Rssi)
+			CollectWirelessMetrics(c.omadaClientTrafficDown, prometheus.CounterValue, item.TrafficDown)
+			CollectWirelessMetrics(c.omadaClientTrafficUp, prometheus.CounterValue, item.TrafficUp)
+			CollectWirelessMetrics(c.omadaClientTxRate, prometheus.GaugeValue, item.TxRate)
+			CollectWirelessMetrics(c.omadaClientRxRate, prometheus.GaugeValue, item.RxRate)
+
 			ch <- prometheus.MustNewConstMetric(c.omadaClientDownloadActivityBytes, prometheus.GaugeValue, item.Activity,
-				item.HostName, item.Vendor, "", "", item.Ip, item.Mac, site, client.SiteId, item.ApName, item.Ssid, wifiMode)
+				item.HostName, item.Vendor, item.Ip, item.Mac, site, client.SiteId, "wireless", item.ApName, item.Ssid, wifiMode, connectionModeFmt, "", "")
 
-			ch <- prometheus.MustNewConstMetric(c.omadaClientSignalDbm, prometheus.GaugeValue, -item.SignalLevel,
-				item.HostName, item.Vendor, item.Ip, item.Mac, item.ApName, site, client.SiteId, item.Ssid, wifiMode)
 		}
 		if !item.Wireless {
+			connectionModeFmt = "wired"
 			ch <- prometheus.MustNewConstMetric(c.omadaClientDownloadActivityBytes, prometheus.GaugeValue, item.Activity,
-				item.HostName, item.Vendor, port, vlanId, item.Ip, item.Mac, site, client.SiteId, "", "", "")
+				item.HostName, item.Vendor, item.Ip, item.Mac, site, client.SiteId, "wired", "", "", "", "", port, vlanId)
+		}
+		totals[connectionModeFmt] += 1
+	}
+
+	for connectionModeFmt := range totals {
```

Comment:
You can iterate over the map to get the key=`k` and value=`v` instead of having to parse them like this: `float64(totals[connectionModeFmt])`
```suggestion
	for k, v := range totals {
```

Reply:
Thanks, I hardly write Go but should have realized this should be possible